src/compute-client/src/controller.rs

```diff
@@ -328,6 +329,26 @@ impl<T> ComputeController<T> {
             storage,
         }
     }
+
+    /// Remove orphaned compute replicas from the orchestrator. These are replicas that the
+    /// orchestrator is aware of, but not the controller.
+    pub async fn remove_orphans(&self) -> Result<(), RemoveOrphansError> {
+        // This method must be called only after we have loaded the catalog state. Assert
+        // that this has happened by checking that we have at least one cluster (there should
+        // always be at least the system cluster)
+        assert!(self.instances.len() > 0);
```

Comment:
It's to ensure we populate the controller properly before calling remove orphans. If that happens, it would be quite dramatic, as `remove_orphans` would then go off to delete all replicas. It's removed in the new PR as it works quite differently now.